app/src/main/java/com/github/se/project/ui/components/LessonLocationDisplay.kt

```diff
@@ -23,33 +49,130 @@ fun LessonLocationDisplay(
-    Text(text = "Lesson Location", style = MaterialTheme.typography.titleMedium)
-
-    Card(
-        modifier = Modifier.fillMaxWidth(),
-        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)) {
-          Box(modifier = Modifier.fillMaxWidth().height(200.dp)) {
-            GoogleMap(
-                modifier = Modifier.fillMaxSize(),
-                cameraPositionState = cameraPositionState,
-                properties = MapProperties(isMyLocationEnabled = false),
-                uiSettings =
-                    MapUiSettings(
-                        zoomControlsEnabled = false,
-                        scrollGesturesEnabled = false,
-                        zoomGesturesEnabled = false,
-                        tiltGesturesEnabled = false,
-                        rotationGesturesEnabled = false,
-                    )) {
-                  Marker(state = MarkerState(position = location), title = lessonTitle)
-                }
-          }
+  // Request for location permission launcher
+  LocationPermissionHandler { location ->
+    userLocation = location
+    isLocationChecked = true
+
+    // Update the camera position based on user location availability
+    cameraPositionState.position =
+        if (location != null) {
+          getCameraPositionForBothLocations(
+              userLocation = location, lessonLocation = lessonLocation)
+        } else {
+          CameraPosition.fromLatLngZoom(lessonLocation, 10f)
         }
   }
+
+  if (isLocationChecked) {
+    Column(modifier = modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+      Text(text = "Lesson Location", style = MaterialTheme.typography.titleMedium)
+
+      Card(
+          modifier = Modifier.fillMaxWidth(),
+          elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)) {
+            Box(modifier = Modifier.fillMaxWidth().height(200.dp)) {
+              GoogleMap(
+                  modifier = Modifier.fillMaxSize(),
+                  cameraPositionState = cameraPositionState,
+                  properties = MapProperties(isMyLocationEnabled = (userLocation != null)),
+                  uiSettings =
+                      MapUiSettings(
+                          zoomControlsEnabled = false,
+                          scrollGesturesEnabled = false,
+                          zoomGesturesEnabled = false,
+                          tiltGesturesEnabled = false,
+                          rotationGesturesEnabled = false)) {
+                    // Marker for the lesson location
+                    Marker(state = MarkerState(position = lessonLocation), title = lessonTitle)
+                  }
+            }
+          }
+    }
+  }
+}
+
+/**
+ * Utility function to calculate the bounds to show both the user's location and lesson location,
+ * and update the camera position to fit them.
+ */
+fun getCameraPositionForBothLocations(
```

Comment:
Having a function to fit both points in automatically is great! From some manual testing though, it feels a little more zoomed-out than necessary